lib/CMakeLists.txt

```diff
@@ -225,7 +225,7 @@ SET_TARGET_PROPERTIES(zip PROPERTIES VERSION 5.0 SOVERSION 5)
 ENDIF()
 
 
-TARGET_LINK_LIBRARIES(zip ${ZLIB_LIBRARIES} ${OPTIONAL_LIBRARY})
+TARGET_LINK_LIBRARIES(zip PUBLIC ${ZLIB_LIBRARIES} ${OPTIONAL_LIBRARY})
```

Comment:
Just for clarification: Are the (compression) libraries that libzip depends on really `PUBLIC`? None of their symbols are passed on to the outside via header files.

Reply:
As I understand it, PUBLIC mirrors the previous behaviour.

If symbols from the other libraries aren't visible from outside, then it sounds like `PRIVATE` might be the more correct way to go, but I'm not familiar enough with how this works to say for sure.

Reply:
> As I understand it, PUBLIC mirrors the previous behaviour.

Correct. But judging from the usage those libraries are only internally used, so they can be PRIVATE. Simple rule of thumb: If a dependent libraries headers are included in this libraries "public headers" (expected to be used by users), then use PUBLIC, else PRIVATE.